llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp

```diff
@@ -21,11 +21,18 @@ using namespace llvm;
 void llvm::getSYCLDeviceRequirements(
     const module_split::ModuleDesc &MD,
     std::map<StringRef, util::PropertyValue> &Requirements) {
-  auto ExtractIntegerFromMDNodeOperand = [=](const MDNode *N,
-                                             unsigned OpNo) -> int32_t {
+  auto ExtractSignedIntegerFromMDNodeOperand = [=](const MDNode *N,
+                                                   unsigned OpNo) -> int64_t {
```

Comment:
Do we really need 64-bit integers here?

Reply:
getSExtValue()  returns int64_t https://llvm.org/doxygen/classllvm_1_1ConstantInt.html#aa8f4be0661aa64f5b1f20b15e93bb403, so that seemed logically correct to me, otherwise we will need cast to smaller type. 
Also removed unnecessary static_cast. from both lambdas.